### PR TITLE
[tests] Add cron job to update gatsby fixtures

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
PR to add a cron job and script that automatically updates gatsby fixture versions.

### CODE REVIEW

**Comment by Ethan-Arrowood** [comment]
Another option to doing the whole `npm view` check is to use: `npm i gatsby@<tag> --save-exact --package-lock-only` which will update the `gatsby` package version to the right semver value (i.e. `4.24.4`) and update the package lock file all in one...without actually installing all the dependencies I think

File: utils/update-gatsby-fixtures.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+const path = require('path');
```
Comment (TooTallNate) [inline utils/update-gatsby-fixtures.js:1]: Can we do TypeScript for these files and invoke with `ts-node`?

Comment (Ethan-Arrowood) [reply utils/update-gatsby-fixtures.js:1]: The github action thing we use to execute doesn't have typescript support it seems https://github.com/actions/github-script/issues/294

Is this a blocker? I we use this pattern in vercel/api as well as the other cron scripts here in this repo so figuring out typescript support seems a bit out of scope

File: utils/update-gatsby-fixtures.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+/**
+ * invoked by ../.github/workflows/cron-test-and-update-gatsby.yml
+ * @param {{ github: ReturnType<import('@actions/github').getOctokit>, context: import('@actions/github').context }} param0 Defined by https://github.com/actions/github-script
+ * @returns
+ */
+module.exports = async ({ github, context }) => {
+  exec('git', ['config', '--global', 'user.email', 'infra+release@vercel.com']);
+  exec('git', ['config', '--global', 'user.name', 'vercel-release-bot']);
+  exec('git', ['checkout', branch]);
```
Comment (TooTallNate) [inline utils/update-gatsby-fixtures.js]:
```suggestion
  try {
    // Branch may exist if there's already an existing PR
    exec('git', ['checkout', branch]);
  } catch {
    exec('git', ['checkout', '-b', branch]);
  }
```

File: utils/update-gatsby-fixtures.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        `gatsby version ${newVersion} did not change for fixture ${fixture}, skipping update.`
+      );
+      continue;
+    }
+
+    packageJSON.dependencies.gatsby = newVersion;
+
+    fs.writeFileSync(
+      packageJSONPath,
+      JSON.stringify(packageJSON, null, 2),
```
Comment (TooTallNate) [inline utils/update-gatsby-fixtures.js]:
```suggestion
      JSON.stringify(packageJSON, null, 2) + '\n',
```

File: utils/update-gatsby-fixtures.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    );
+
+    // update lockfiles
+    if (fixture.includes('pnpm')) {
+      exec('pnpm', ['install', '--lockfile-only']);
+    } else {
+      exec('npm', ['install', '--package-lock-only']);
+    }
+  }
+
```
Comment (TooTallNate) [inline utils/update-gatsby-fixtures.js:77]: Should check if there were any updates here, and bail if not.

File: utils/update-gatsby-fixtures.js (hunk 2 above)
Comment (EndangeredMassa) [reply utils/update-gatsby-fixtures.js]: If there's an existing branch, do we need to merge main before continuing?

**Review by TooTallNate — APPROVED** [review]

File: utils/update-gatsby-fixtures.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+ * invoked by ../.github/workflows/cron-test-and-update-gatsby.yml
+ * @param {{ github: ReturnType<import('@actions/github').getOctokit>, context: import('@actions/github').context }} param0 Defined by https://github.com/actions/github-script
+ * @returns
+ */
+module.exports = async ({ github, context }) => {
+  exec('git', ['config', '--global', 'user.email', 'infra+release@vercel.com']);
+  exec('git', ['config', '--global', 'user.name', 'vercel-release-bot']);
+  try {
+    // Branch may exist if there's already an existing PR
+    exec('git', ['checkout', branch]);
```
Comment (EndangeredMassa) [inline utils/update-gatsby-fixtures.js:17]: **question (non-blocking):** should we also merge main in this case?

Comment (Ethan-Arrowood) [reply utils/update-gatsby-fixtures.js:17]: i assume itd happen automatically cause of automerge or our manual input. if it becomes an issue Ill modify

**Pull request merged**